### PR TITLE
Feat - Add URL validation utility function

### LINKED ISSUE #7: Create URL validation utility
Create a utility function to validate URLs. This should check if a given string is a properly formatted URL (http/https protocols), handle edge cases, and include JSDoc comments.

### PR DESCRIPTION
**Pull request by raphaeltx**
## Description
* Add URL validation utility with additional unit tests;
* Install vitest dependency to unit tests;
* Update temperature.test.js imports.

## Related Issue
Fixes #7

## Type of Change
- [ ] Bug fix
- [x] New feature
- [x] Documentation update
- [ ] Code refactoring

## Testing
- [x] Tests pass locally
- [x] Added new tests for new functionality
- [ ] Manual testing completed

## Checklist
- [x] Code follows project style guidelines
- [x] Self-review completed
- [x] Comments added for complex code
- [ ] Documentation updated
- [ ] No breaking changes

### CODE REVIEW

File: tests/utilities/url.test.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      expect(isURLValid("http://127.0.0.1")).toBe(true);
+    });
+
+    test("should return true for URL with path", () => {
+      expect(isURLValid("https://example.com/path/to/page")).toBe(true);
+    });
+
+    test("should return true for URL with query params", () => {
+      expect(isURLValid("https://example.com?param=value")).toBe(true);
+    });
```
[the remark this answers is not in the document]
Comment (raphaeltx) [reply tests/utilities/url.test.js:68]: Done! Added some tests with the Anchor and some variations (with path, params, encoded, etc).

File: tests/utilities/url.test.js
[the remark this answers is not in the document]
Comment (LeftShotDev) [reply tests/utilities/url.test.js:1]: All the changes look great. You're good to merge.

[the remark this answers is not in the document]
Comment (raphaeltx) [reply tests/utilities/url.test.js:1]: Awsome! Do we have permission to merge ourselves?

[the remark this answers is not in the document]
Comment (raphaeltx) [reply tests/utilities/url.test.js:1]: Yeah i don't have permission to merge
